build/common.sh

```diff
@@ -242,11 +242,15 @@ function kube::build::is_osx() {
   [[ "$(uname)" == "Darwin" ]]
 }
 
+function kube::build::is_gnu_sed() {
+  [[ $(sed --version 2>&1) == *GNU* ]]
```

Comment:
Hi @jbeda , yes, I had tested on OS X stock `sed` and it worked correctly. Sorry for the slow reply; I had to do some digging to understand _why_ it worked. 😄 

`$()` command substitution passes on the `-1` exit code from OS X BSD sed. The answer, according to http://www.tldp.org/LDP/abs/html/options.html#OPTIONSREF, is that `set -e` will _Abort script at first error, when a command exits with non-zero status (**except in until or while loops, if-tests, list constructs**)_. So `[[ $(sed --version 2>&1) == *GNU* ]]` will return an error code but it is harmless, even when `set -e` is in play. Here's a minimal demonstration. Note that the script exits and the final echo is never printed.

```
bash-3.2$ cat test.sh
#!/bin/bash

[[ $(false) ]]
echo "Extended test false: $?"
$(false)
echo "Bare false: $?"

echo set -e
set -e

[[ $(false) ]]
echo "Extended test false: $?"
$(false)
echo "Bare false: $?"
```
```
bash-3.2$ ./test.sh
Extended test false: 1
Bare false: 1
set -e
Extended test false: 1
```